spec/defines/fail2ban_jail_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe 'fail2ban::jail' do
+  let(:title) { 'spec_test_jail' }
+  let(:pre_condition) { 'include fail2ban' }
+  let(:facts) do
+    {
```

Comment:
instead of mocking the facts by hand, please use rspec-puppet-facts. An example: https://github.com/voxpupuli/puppet-zabbix/blob/master/spec/classes/web_spec.rb#L18

docs: https://github.com/camptocamp/rspec-puppet-facts